reconcile/openshift_resources_base.py

```diff
@@ -1000,13 +1011,33 @@ def run(
     internal=None,
     use_jump_host=True,
     providers=None,
-    cluster_name=None,
+    cluster_name: Optional[Sequence[str]] = None,
+    exclude_cluster: Optional[Sequence[str]] = None,
     namespace_name=None,
     init_api_resources=False,
     defer=None,
 ):
+    # https://click.palletsprojects.com/en/8.1.x/options/#multiple-options
+    cluster_names = cluster_name
+    exclude_clusters = exclude_cluster
+
+    if exclude_cluster and not dry_run:
+        raise RuntimeError("--exclude-cluster is only supported in dry-run mode")
+
+    if exclude_cluster and cluster_name:
+        raise RuntimeError(
+            "--cluster-name and --exclude-cluster can not be used together"
+        )
+    if cluster_names and len(cluster_names) > 1 and not dry_run:
+        raise RuntimeError(
+            "Running with multiple clusters is only supported in dry-run mode"
+        )
```

Comment:
Could you add tests for those exceptions? They are fairly simple to test.

Reply:
Added a test for this. PTAL